WPMedia/fusion-news-theme-blocks
# Tmedia 575 social profile

## Description
User Story: As a user, I want to view and connect or disconnect my social profiles on the Account page.

## Jira Ticket
- [TMEDIA-575](https://arcpublishing.atlassian.net/browse/TMEDIA-575)

## Acceptance Criteria
Add custom field to block (TMEDIA-578) to enable SocialProfile

If unchecked no social profile management is shown

If checked:

Show social profile connect functionality 

Show “Connected Accounts” heading string (translated)

User can connect a social account to an existing account

User can disconnect a social account as long another login is set (either email/password or the other social)

## Test Steps

1. Checkout this branch `git checkout TMEDIA-575-social-profile`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/identity-block`
3. Create an account that matches a gmail account you have in the sign up screen
4. Login as that user. 
5. Then go to the account management page. You should enable the social settings for that page in the editor. 
6. Click connect google button. 
7. You should see the page refresh and login as that user (if it hasn't been created before) and you will now see connected as that user
8. Click "Disconnect" to disconnect from that user for the password-user email 
9. Create a new email and sign in that doesn't match the gmail 
10. Delete in the user the gmail account 
11. Confirm the deletion in requests
12. Sign in with that gmail account from your other xx@washpost.com account. 
13. You should be connected in gmail with that user. 

## Effect Of Changes
### Before
- no social screen 

### After
designs based on conversation with Jenae and Nick's poc 

![Screen Shot 2021-12-08 at 15 04 40](https://user-images.githubusercontent.com/5950956/145292639-572602c4-aa7c-45bc-a48d-c1bf26c318af.png)

## Dependencies or Side Effects
- ensure you have the mod header extension set for testing 
- facebook can't be tested locally 

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
